### PR TITLE
Fix the statuscode for switch to nonexistent service

### PR DESCRIPTION
**Pull request by pablothedude**
Change the thrown exception to be a 404 instead a 500. Also renamed
the getServiceId method to be more descriptive. Also added a dedicated
symfony form to switch services to have csrf support.

### CODE REVIEW

File: tests/webtests/ServiceOverviewTest.php [hunk 1]
```diff
@@ -47,11 +43,12 @@ public function test_users_can_use_the_service_overview_page()
         $serviceRepository = $this->getServiceRepository();
         $surfNet = $serviceRepository->findByName('SURFnet');
 
+        $this->logIn('ROLE_USER', [$surfNet]);
+        //$this->switchToService('SURFnet');
```
Comment (MKodde) [inline tests/webtests/ServiceOverviewTest.php]: Development leftover?

File: src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Service/AuthorizationService.php [hunk 2]
```diff
@@ -232,7 +232,7 @@ function ($id) use ($contact) {
      * @return \Surfnet\ServiceProviderDashboard\Domain\Entity\Service
      * @throws ServiceNotFoundException
      */
-    public function getServiceById($serviceId)
+    public function changeActiveService($serviceId)
```
[the remark this answers is not in the document]
Comment (pablothedude) [reply src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Service/AuthorizationService.php]: I agree that it's not optimal but I found it more intuitive that the method changes the active service and also returns an Service entity then a getter which eventually will change the active service. Maybe we should change this into account if the admin page will get better search capabilities as this is an ongoing wish.

File: src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Form/Service/ServiceSwitcherType.php [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+use Surfnet\ServiceProviderDashboard\Infrastructure\DashboardBundle\Command\Service\SelectServiceCommand;
+use Surfnet\ServiceProviderDashboard\Infrastructure\DashboardBundle\Service\AuthorizationService;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * The service switcher type form
```
[the remark this answers is not in the document]
Comment (pablothedude) [reply src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Form/Service/ServiceSwitcherType.php]: Removed

File: tests/webtests/ServiceOverviewTest.php (hunk 1 above)
Comment (pablothedude) [reply tests/webtests/ServiceOverviewTest.php]: Removed

**Pull request merged**